<?xml version="1.0" ?>
<sdf version="1.6">
  <-model name="x500_ir">
    <link name="base_link">
      <inertial>
        <mass>1.5</mass>
        <inertia>
          <ixx>0.02</ixx>
          <iyy>0.02</iyy>
          <izz>0.04</izz>
        </inertia>
      </inertial>
      <visual name="base_visual">
        <geometry>
          <box>
            <size>0.2 0.2 0.1</size>
          </box>
        </geometry>
        <material>
          <script>
            <name>Gazebo/Grey</name>
          </script>
        </material>
      </visual>
      <collision name="base_collision">
        <geometry>
          <box>
            <size>0.2 0.2 0.1</size>
          </box>
        </geometry>
      </collision>
    </link>

    <link name="ir_camera_link">
      <pose>0.1 0 0.05 0 0 0</pose>
      <inertial>
        <mass>0.01</mass>
        <inertia>
          <ixx>1e-6</ixx>
          <iyy>1e-6</iyy>
          <izz>1e-6</izz>
        </inertia>
      </inertial>
    </link>
    <joint name="ir_camera_joint" type="fixed">
      <parent>base_link</parent>
      <child>ir_camera_link</child>
    </joint>
    <sensor name="ir_camera" type="camera">
      <pose>0 0 0 0 0 0</pose>
      <camera>
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>640</width>
          <height>480</height>
          <format>R8</format>
        </image>
        <clip>
          <near>0.1</near>
          <far>100</far>
        </clip>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <ros>
          <namespace>/x500_ir</namespace>
          <remapping>image_raw:=ir/image_raw</remapping>
          <remapping>camera_info:=ir/camera_info</remapping>
        </ros>
        <frame_name>ir_camera_link</frame_name>
      </plugin>
    </sensor>

    <link name="gps_link">
      <pose>0 0 0.1 0 0 0</pose>
      <inertial>
        <mass>0.01</mass>
        <inertia>
          <ixx>1e-6</ixx>
          <iyy>1e-6</iyy>
          <izz>1e-6</izz>
        </inertia>
      </inertial>
    </link>
    <joint name="gps_joint" type="fixed">
      <parent>base_link</parent>
      <child>gps_link</child>
    </joint>
    <sensor name="gps" type="gps">
      <plugin name="gps_plugin" filename="libgazebo_ros_gps.so">
        <ros>
          <namespace>/x500_ir</namespace>
          <remapping>fix:=gps/fix</remapping>
        </ros>
      </plugin>
    </sensor>
  </model>
</sdf>